### DISCUSSION TITLE
flip middleware not respecting viewport when positioning a HTML popover

### DISCUSSION THREAD

**Discussion by ollicle** [floating-ui/floating-ui#2780] [General]
When the floating element has a [popover](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/popover) attribute the overflow detection appears to be failing.

Demo: https://agate-equal-tang.glitch.me
Demo code: https://glitch.com/edit/#!/agate-equal-tang

Scroll the page up/down to place the buttons close to the top or bottom of the viewport before clicking the buttons.

The non popover element is positioned as expected – avoiding the top/bottom of the viewport.
However the popover element is not – cropped by the top of the viewport when the buttons are near the top.

Any suggestions to work around this?

**Comment by atomiks** [comment]
Seems like `fixed` strategy works but not `absolute`. So looks like a bug, but does have a workaround.
Edit: you can make an issue for that
